src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/YFlowValidatorTest.java

```diff
@@ -222,6 +222,68 @@ public void failIfNegativeBandwidthProvided()
         yFlowValidator.validate(request);
     }
 
+    @Test(expected = InvalidFlowException.class)
+    public void failIfMaxLatencyTier2HigherThanMaxLatency()
+            throws InvalidFlowException, UnavailableFlowEndpointException {
+        YFlowRequest request = YFlowRequest.builder()
+                .yFlowId("test")
+                .maxLatency((long) 1000)
+                .maxLatencyTier2((long) 500)
+                .sharedEndpoint(FlowEndpoint.builder()
+                        .switchId(SWITCH_ID_1)
+                        .portNumber(PORT_1)
+                        .build())
+                .subFlows(Arrays.asList(SubFlowDto.builder()
+                                .flowId("test_1")
+                                .sharedEndpoint(new SubFlowSharedEndpointEncapsulation(1, 0))
+                                .endpoint(FlowEndpoint.builder()
+                                        .switchId(SWITCH_ID_2)
+                                        .portNumber(PORT_2)
+                                        .build())
+                                .build(),
+                        SubFlowDto.builder()
+                                .flowId("test_2")
+                                .sharedEndpoint(new SubFlowSharedEndpointEncapsulation(2, 0))
+                                .endpoint(FlowEndpoint.builder()
+                                        .switchId(SWITCH_ID_3)
+                                        .portNumber(PORT_3)
+                                        .build())
+                                .build()))
+                .build();
+        yFlowValidator.validate(request);
```

Comment:
I think you can move Request creation to some method, with 2 params - maxLatency and maxLatencytier2 to
1. point on main parameters in each test
2. same lines of code

Reply:
done